File: Library/Operations/BruteForcePasswordGenerator.vb
Imports System.Numerics

Namespace Operations.BruteForcePasswordGenerator
    Public Class BruteForcePasswordGenerator
        Public Sub Run()
            GeneratePassword()
        End Sub

        Private Sub GeneratePassword()
            Dim chars() As Char = {"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"}

            Dim passwordLength As Short = 0

            Console.Write("Enter the password length: ")

            passwordLength = Convert.ToInt32(Console.ReadLine())

            ' Double casts were redundant
            Dim iPossibilities As BigInteger = CType(Math.Pow(chars.Length, passwordLength), BigInteger)
            Console.Write("{0} words total. Press ENTER to continue;", iPossibilities)
            Console.ReadLine()

            Dim watch As Stopwatch = Stopwatch.StartNew()

            For i As BigInteger = 0 To iPossibilities - 1
                Dim theWord As String = ""
                Dim val As BigInteger = i
                For j As Short = 0 To passwordLength - 1
                    Dim ch As BigInteger = (val Mod chars.Length)
                    theWord = chars(CType(ch, Short)) + theWord
                    val = val / chars.Length
                Next

                Console.WriteLine(theWord)
            Next

            watch.Stop()
            Dim elapsedMs As Long = watch.ElapsedMilliseconds
            Console.WriteLine("It took {0} milliseconds ({1} seconds) to generate {2} possible combinations", elapsedMs, elapsedMs / 1000, iPossibilities)
        End Sub
    End Class
End Namespace

File: CS 317/Operations/BruteForcePasswordGenerator/BruteForcePasswordGenerator.cs
using System;
using System.Diagnostics;
using System.Numerics;

namespace CS317Program.Operations.BruteForcePasswordGenerator
{
    class BruteForcePasswordGenerator
    {
        public void Run()
        {
            GeneratePassword();
        }

        private void GeneratePassword()
        {
            char[] chars = { 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z' };


            int passwordLength = 0;

            Console.WriteLine("Enter the password length");

            passwordLength = Convert.ToInt32(Console.ReadLine());

            BigInteger iPossibilities = (BigInteger)Math.Pow(chars.Length, passwordLength);
            Console.WriteLine("{0} words total. Press enter to continue;", iPossibilities);
            Console.ReadLine();

            var watch = Stopwatch.StartNew();
            for (BigInteger i = 0; i < iPossibilities; i++)
            {
                string theword = "";
                BigInteger val = i;
                for (int j = 0; j < passwordLength; j++)
                {
                    BigInteger ch = val % chars.Length;
                    theword = chars[(int)ch] + theword;
                    val = val / chars.Length;
                }
                Console.WriteLine(theword);
            }
            watch.Stop();
            var elapsedMs = watch.ElapsedMilliseconds;
            Console.WriteLine("It took {0} seconds to generate {1} possible combinations", elapsedMs / 1000, iPossibilities);
        }
    }
}
